job/service.go

```diff
@@ -848,19 +855,30 @@ func (srv *Service) resolveDependency(ctx context.Context, projectSpec models.Pr
 	srv.notifyProgress(progressObserver, &models.ProgressJobDependencyResolutionFinished{})
 }
 
-func (srv *Service) resolveAndPersist(ctx context.Context, currentSpec models.JobSpec, projectSpec models.ProjectSpec, progressObserver progress.Observer) (interface{}, error) {
+func (srv *Service) resolve(ctx context.Context, currentSpec models.JobSpec, projectSpec models.ProjectSpec) ([]string, error) {
 	var err error
 	if currentSpec.Assets, err = srv.assetCompiler(currentSpec, srv.Now()); err != nil {
-		return currentSpec.Name, fmt.Errorf("%w: %s", errAssetCompilation, err.Error())
+		return nil, fmt.Errorf("%w: %s", errAssetCompilation, err.Error())
 	}
-	resolvedSpec, err := srv.dependencyResolver.Resolve(ctx, projectSpec, currentSpec, progressObserver)
+	jobSourceURNs, err := srv.dependencyResolver.ResolveInferredDependencies(ctx, projectSpec, currentSpec)
```

Comment:
We can call the `pluginService` directly from here, not through `dependencyResolver`